processor/processor.go

```diff
@@ -1490,8 +1492,7 @@ func (proc *Handle) processJobsForDest(partition string, subJobs subJob) *transf
 			},
 			)
 			if proc.config.eventSchemaV2Enabled && // schemas enabled
-				// source has schemas enabled or if we override schemas for all sources
-				(source.EventSchemasEnabled || proc.config.eventSchemaV2AllSources) &&
+				proc.config.eventAuditEnabled[batchEvent.WorkspaceId] &&
```

Comment:
You need to acquire a read lock before reading the value from `eventAuditEnabled` here